src/error.rs

```diff
@@ -2,14 +2,11 @@ use crate::launch::ProcfileConversionError;
 use crate::procfile::ProcfileParsingError;
 use indoc::formatdoc;
 
-#[derive(thiserror::Error, Debug)]
+#[derive(Debug)]
 pub enum ProcfileBuildpackError {
-    #[error("Cannot read Procfile contents: {0}")]
     CannotReadProcfileContents(std::io::Error),
-    #[error("Procfile parsing error: {0}")]
-    ProcfileParsingError(#[from] ProcfileParsingError),
-    #[error("Procfile conversion error: {0}")]
-    ProcfileConversionError(#[from] ProcfileConversionError),
+    Parsing(ProcfileParsingError),
```

Comment:
So for my Python CNB I've tried to avoid redundancy in the enum variant names where possible. The enum type itself is named `ProcfileBuildpackError`, so by definition we know it's Procfile related, and also they are all errors. Having shorter names then makes the nested `match`es slightly more readable (all of the variables starting with `Procfile...` were blurring into one).

However I can also see that having the names match may be appealing too.